memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -228,6 +227,47 @@ object MutableHistogram {
   }
 }
 
+/**
+ * MaxHistogram improves quantile calculation accuracy with a known max value recorded from the client.
+ * Whereas normally Prom histograms have +Inf as the highest bucket, and we cannot interpolate above the last
+ * non-Inf bucket, having a max allows us to interpolate from the rank up to the max.
+ * When the max value is lower, we interpolate between the bottom of the bucket and the max value.
+ * Both changes mean that the 0.90+ quantiles return much closer to the max value, instead of interpolating or clipping.
+ * The quantile result can never be above max, regardless of the bucket scheme.
+ */
+final case class MaxHistogram(innerHist: MutableHistogram, max: Double) extends HistogramWithBuckets {
+  final def buckets: HistogramBuckets = innerHist.buckets
+  final def bucketValue(no: Int): Double = innerHist.bucketValue(no)
+
+  def serialize(intoBuf: Option[MutableDirectBuffer] = None): MutableDirectBuffer = ???
```

Comment:
No, the serialization of the max value is not needed since this class is only used to compute the quantile.  You can't ever get a `MaxHistogram` result.

I thought of serializing and taking a different approach but it introduces more complexities in the lower layer that I'd rather not deal with.